File: Lab 3/Bank Account/SavingsAccount.java
import java.util.*;


class SavingsAccount extends BankAccount{
	private double interestRate;

	public SavingsAccount(String n, double b, double ir){
		this.interestRate = ir;
		this.setName(n);
		this.setBalance(b);;
	}

	public void addInterest(){
		double interest = this.getBalance() * (this.interestRate/100);
		double temp = this.getBalance() + interest;
		this.setBalance(temp);
	}
}

File: Lab7/SavingsAccount.cpp
//
// Created by Vasanth Rajasekaran on 11/25/18.
//

#include "SavingsAccount.h"
#include "BankAccount.h"

void SavingsAccount::addInterest(){ // adds interest based on the balance
    double interest = BankAccount::getBalance() * (interestRate/100);
    double temp = BankAccount::getBalance() + interest;
    balance = temp;
}

SavingsAccount::SavingsAccount(){ // default constructor
    name = "";
    balance=0;
    interestRate = 0;
};
SavingsAccount::SavingsAccount(string n, double b, double i){ // constructor with all the components
    name = n;
    balance = b;
    interestRate = i;
};
